server/responseTypes.ts

```diff
@@ -0,0 +1,37 @@
+export interface APIPageResponse<T> {
+    count: number,
+    next:string | null | undefined,
+    previous: string | null | undefined, 
+    results: T[]
+}
+
+export interface APIPokemon {
+    pokemonName: string, 
+    pokemonNumber: string, 
+    pokemonTypes: APIPokemonType[], 
+    boxBg: string, 
+    pokemonSvg: string
+}
+
+export enum APIPokemonType {
+    Bug = 'Bug',
+    Dark = 'Dark',
+    Dragon = 'Dragon',
+    Electric = 'Electric',
+    Fairy = 'Fairy',
+    Fighting = 'Fighting',
+    Fire = 'Fire',
+    Flying = 'Flying',
+    Ghost = 'Ghost',
+    Grass = 'Grass',
+    Ground = 'Ground',
+    Ice = 'Ice',
+    Normal = 'Normal',
+    Poison = 'Poison',
+    Psychic = 'Psychic',
+    Rock = 'Rock',
+    Steel = 'Steel',
+    Water = 'Water',
+
+    Unknown = 'Unknown'
+  }
```

Comment:
If you use `APIPokemonType` inside `APIPokemon` please move `APIPokemonType` before `APIPokemon`.

```suggestion
export enum APIPokemonType {
    Bug = 'Bug',
    Dark = 'Dark',
    Dragon = 'Dragon',
    Electric = 'Electric',
    Fairy = 'Fairy',
    Fighting = 'Fighting',
    Fire = 'Fire',
    Flying = 'Flying',
    Ghost = 'Ghost',
    Grass = 'Grass',
    Ground = 'Ground',
    Ice = 'Ice',
    Normal = 'Normal',
    Poison = 'Poison',
    Psychic = 'Psychic',
    Rock = 'Rock',
    Steel = 'Steel',
    Water = 'Water',

    Unknown = 'Unknown'
  }
  
  export interface APIPokemon {
    pokemonName: string, 
    pokemonNumber: string, 
    pokemonTypes: APIPokemonType[], 
    boxBg: string, 
    pokemonSvg: string
}
```